ts-relayer-tests/src/ics721.spec.ts

```diff
@@ -311,95 +391,366 @@ test.serial(
+    timeout: {
+      block: {
+        revision: 1,
+        height: 90000,
+      },
+    },
+  };
+  transferResponse = await sendNft(
+    wasmClient,
+    wasmCw721,
+    wasmCw721OutgoingProxy,
+    ibcMsg,
+    tokenId
+  );
+  t.truthy(transferResponse);
+
+  // Relay and verify we got an error
+  t.log("relaying packets");
+  info = await otherChannel.link.relayAll();
+  assertAckErrors(info.acksFromA);
+  // assert no change before and after relay
+  const afterWasmOutgoingClassTokenToChannelList = await outgoingChannels(
+    wasmClient,
+    wasmIcs721
+  );
+  const afterWasmIncomingClassTokenToChannelList = await incomingChannels(
+    wasmClient,
+    wasmIcs721
+  );
+  const afterWasmNftContractsToClassIdList = await nftContracts(
+    wasmClient,
+    wasmIcs721
+  );
+  t.deepEqual(
+    beforeWasmOutgoingClassTokenToChannelList,
+    afterWasmOutgoingClassTokenToChannelList,
+    `outgoing channels must be unchanged:
+- wasm before: ${JSON.stringify(beforeWasmOutgoingClassTokenToChannelList)}
+- wasm after: ${JSON.stringify(afterWasmOutgoingClassTokenToChannelList)}`
+  );
+  t.deepEqual(
+    beforeWasmIncomingClassTokenToChannelList,
+    afterWasmIncomingClassTokenToChannelList,
+    `incoming channels must be unchanged:
+- wasm before: ${JSON.stringify(beforeWasmIncomingClassTokenToChannelList)}
+- wasm after: ${JSON.stringify(afterWasmIncomingClassTokenToChannelList)}`
+  );
+  t.deepEqual(
+    beforeWasmNftContractsToClassIdList,
+    afterWasmNftContractsToClassIdList,
+    `nft contracts must be unchanged:
+- wasm before: ${JSON.stringify(beforeWasmNftContractsToClassIdList)}
+- wasm after: ${JSON.stringify(afterWasmNftContractsToClassIdList)}`
+  );
+  const afterOsmoOutgoingClassTokenToChannelList = await outgoingChannels(
+    osmoClient,
+    osmoIcs721
+  );
+  const afterOsmoIncomingClassTokenToChannelList = await incomingChannels(
+    osmoClient,
+    osmoIcs721
+  );
+  const afterOsmoNftContractsToClassIdList = await nftContracts(
+    osmoClient,
+    osmoIcs721
+  );
+  t.deepEqual(
+    beforeOsmoOutgoingClassTokenToChannelList,
+    afterOsmoOutgoingClassTokenToChannelList,
+    `outgoing channels must be unchanged:
+- osmo before: ${JSON.stringify(beforeOsmoOutgoingClassTokenToChannelList)}
+- osmo after: ${JSON.stringify(afterOsmoOutgoingClassTokenToChannelList)}`
+  );
+  t.deepEqual(
+    beforeOsmoIncomingClassTokenToChannelList,
+    afterOsmoIncomingClassTokenToChannelList,
+    `incoming channels must be unchanged:
+- osmo before: ${JSON.stringify(beforeOsmoIncomingClassTokenToChannelList)}
+- osmo after: ${JSON.stringify(afterOsmoIncomingClassTokenToChannelList)}`
+  );
+  t.deepEqual(
+    beforeOsmoNftContractsToClassIdList,
+    afterOsmoNftContractsToClassIdList,
+    `nft contracts must be unchanged:
+- osmo before: ${JSON.stringify(beforeOsmoNftContractsToClassIdList)}
+- osmo after: ${JSON.stringify(afterOsmoNftContractsToClassIdList)}`
+  );
```

Comment:
This is the ts relayer test, checking states before and after a failed back transfer